### ISSUE TITLE
DrawRotatedText is glyph-reversed, letters drawn back to front in 3.1.6, but is OK in 3.1.5

### ISSUE THREAD

**Issue by betrixed** [wxWidgets/wxWidgets#22278]
wxWidgets 3.1.6  for wxPaintDC, DrawRotatedText, eg at -90 degrees angle, draws the font letters as mirror image reversed, with correct letter order but each font glyph is drawn back to front. This occurs for wxGTK3 on linux debian, plasma or xfce4 desktop on AMD64.

Platform Debian 11 - bullseye
DrawRotatedText works properly for  wxWidgets 3.1.5 compiled in the same environment (either ARM64 or AMD64),
but draws letters reversed for 3.1.6.

**Comment by paulcor** [comment]
Change in behavior is due to eafa2481175976fca667d242aed547bb19a65b88

**Comment by betrixed** [comment]
Finally I set up a debug version of 3.1.6 and stepped through src/gtk/dc.cpp   

Yes the complicated changes to void wxGTKCairoDCImpl::DoDrawText causes the issue.
I reverted my source version of that function that found in 3.1.5.
It seems to me that the strategy of detecting "mirror inversion" by comparison of coordinate transform of dx and dy, is not going to work for both rotation and mirror transformations, because both will detect that "yinverted" is true because dy < 0.
The -90 rotation situation, and therefore other rotations, will wrongly apply additional scale and coordinate offset.
A strategy according to instant web search of drawing rotated text in Cairo is - 
1. saveState.
2. set a transform, apply rotation or mirror scaling,
3. call DrawText, which assumes everything is normal,  does not try detecting what the underlying transform matrix is. The transform does the work.
4. restore State.

    Transform Matrix with adjustments for x, y, need to be done prior to DrawText call, else we may end up with special case versions of DrawText for specific transforms.    As it is my code (graph vertical axis drawing) already assumes the naive behaviour. 


It seems changes made to DrawText for 3.1.6 break the "not knowing" approach. I am not very interested in finding out what other parts of code  now depend on DrawText detecting "inversion" conditions.
